Sources/SwiftDocCUtilities/CatalogTemplates/CatalogTemplateFactory.swift

```diff
@@ -0,0 +1,61 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021-2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import SwiftDocC
+
+protocol Catalog {
+    var articles: [String : ArticleTemplate] { get }
+    var title: String { get set }
+}
+
+struct CatalogTemplateFactory {
+    
+    enum Error: DescribedError {
+        case missingCatalogTemplate
+        var errorDescription: String {
+            switch self {
+                case .missingCatalogTemplate:
+                return "Provided template name is not a valid catalog template."
+            }
+        }
+    }
+    
```

Comment:
AFAICT this error is never raised anywhere.

```suggestion
```

Reply:
Removed 👍